HEC.FDA.Model/compute/ImpactAreaScenarioSimulation.cs

```diff
@@ -372,7 +372,13 @@ private void ComputeFromStageFrequency(IProvideRandomNumbers randomProvider, IPa
                         {
                             ComputeDamagesFromStageFrequency_WithLevee(randomProvider, frequency_stage, systemResponse_sample, giveMeADamageFrequency, iteration, computeIsDeterministic);
                         }
-                        ComputeLeveePerformance(frequency_stage, systemResponse_sample, iteration);
+                        if (systemResponse_sample.Xvals.Length <= 2)
```

Comment:
This reads like you're checking that a systemResponse sample exists. Is there a chance it could be null? If so, we'd accidentally hit an exception here.